control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -128,6 +152,47 @@ func (r *HostedControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 		return ctrl.Result{}, err
 	}
 
+	// Return early if deleted
+	if !hostedControlPlane.DeletionTimestamp.IsZero() {
+		if err := r.delete(ctx, hostedControlPlane); err != nil {
+			r.Log.Error(err, "failed to delete cluster")
+			return ctrl.Result{}, err
+		}
+		if controllerutil.ContainsFinalizer(hostedControlPlane, finalizer) {
+			controllerutil.RemoveFinalizer(hostedControlPlane, finalizer)
+			if err := r.Update(ctx, hostedControlPlane); err != nil {
+				return ctrl.Result{}, fmt.Errorf("failed to remove finalizer from cluster: %w", err)
+			}
+		}
+		return ctrl.Result{}, nil
+	}
+
+	patchHelper, err := patch.NewHelper(hostedControlPlane, r.Client)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to init patch helper: %w", err)
+	}
+
+	oldStatus := hostedControlPlane.Status.DeepCopy()
+
+	// Sync any status changes during the reconciliation
+	defer func() {
+		availableCondition := getAvailableCondition(hostedControlPlane)
+
+		if len(oldStatus.Conditions) != 0 {
+			oldAvailableCondition := &oldStatus.Conditions[0]
+			if reflect.DeepEqual(oldAvailableCondition, availableCondition) {
```

Comment:
Now that this defer is the single writer of the HostedControlPlane, if `hostedControlPlane.Status.Ready` is changed, but conditions do not change, will `hostedControlPlane.Status.Ready` ever be persisted? (https://github.com/openshift/hypershift/pull/18/files#diff-3d4ffda0834e09a59a4d510523932e20b9234bcb9816e40258a0fb32a87a7006L223-R276)

Reply:
i just noticed this while my lunch was heating up as well.  will change.